.github/workflows/test-and-publish-nightly.yml

```diff
@@ -209,8 +209,20 @@ jobs:
           esac
           echo "jars_directory=jar_artifacts" >> $GITHUB_OUTPUT
           echo "release_directory=$(basename $(find . -type d -iname "async-profiler-*" ))" >> $GITHUB_OUTPUT
+      - name: Setup core dump collection
+        run: |
+          case "${{ matrix.runson.name }}" in
+            macos*)
+              sudo sysctl -w kern.corefile=core.%P
+              sudo sysctl kern.coredump=1
+            ;;
+            *)
+              echo "$GITHUB_WORKSPACE/core" | sudo tee /proc/sys/kernel/core_pattern
+            ;;
+          esac
```

Comment:
Please exclude non-relevant changes from this PR.

Reply:
I added that for debugging: ffdeb52faf34b3c5ca8cc2f07443dd01183a7fb5